### ISSUE TITLE
Invalid APM signature

### ISSUE THREAD

**Issue by doctaweeks** [clalancette/pycdlib#66]
I received an invalid APM signature exception when opening a (common) ISO:

```
iso.open('Fedora-Workstation-Live-x86_64-33-1.2.iso')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/site-packages/pycdlib/pycdlib.py", line 4186, in open
    self._open_fp(fp)
  File "/usr/lib/python3.8/site-packages/pycdlib/pycdlib.py", line 2332, in _open_fp
    if tmp_isohybrid.parse(self._cdfp.read(16 * 2048)):
  File "/usr/lib/python3.8/site-packages/pycdlib/isohybrid.py", line 749, in parse
    self.primary_gpt.parse_primary(instr, self.mac)
  File "/usr/lib/python3.8/site-packages/pycdlib/isohybrid.py", line 505, in parse_primary
    apm_part.parse(instr[offset:])
  File "/usr/lib/python3.8/site-packages/pycdlib/isohybrid.py", line 167, in parse
    raise pycdlibexception.PyCdlibInvalidISO('Invalid APM signature')
pycdlib.pycdlibexception.PyCdlibInvalidISO: Invalid APM signature
```

I have confirmed the file matches upstream hashes. I see the same behavior with a newer Fedora workstation ISO as well (Fedora-Workstation-Live-x86_64-34_Beta-1.3.iso).

**Comment by clalancette** [comment]
Yep, I see this too.  I'll take a look.

**Comment by clalancette** [comment]
So, it looks like the Fedora workstation ISOs (what I am testing against here) are actually buggy.  At least, my understanding of the Apple Partition Map stuff is that offsets 2048, 4096, and 6144 should contain partition maps starting with ASCII 'PM'.  Indeed, if I look back at an earlier Fedora Workstation DVD (Fedora 28 is what I have handy), those all exist.  I think this Fedora 34 ISO wouldn't actually work on an Apple device.

Nevertheless, we want to be able to accept as many ISOs we can, even ones that are buggy.  In this case I'll have to add some detection to see if those sections that I expect are zero, and if so, to skip the APM parsing.

**Comment by clalancette** [comment]
All right, this should be solved by 6f0f7b67835eeb26ed3b045035f9b182d3fa08b0 .  Please give it a whirl and let me know if it fixes the problem for you.  Thanks for the bug report.
